### DISCUSSION TITLE
Does debugging next.js api routes work?

### DISCUSSION THREAD

**Discussion by zksward** [nandorojo/solito#30] [Q&A] [answered]
I'm having an issue getting the next.js routes to debug properly in vs code. When using the withExpo config in next.config.js the files that are available for debugging are transpiled and don't match the source very well. Am I missing something in my configuration to get this working better?

**Comment by nandorojo** [comment, marked as answer]
Hmm interesting, I'm not totally sure what you mean. By debugging, do you mean for errors inside of `pages/api`? Maybe there's a way to ignore that folder(?) I don't use `pages/api`, so I'm not sure about this, but maybe `withExpo` has some unintended behavior of transpiling that folder?

**Reply by zksward** [reply]
First I want to say thanks for the fast response! You've built some really awesome libraries for the community and I'm excited for the future of the projects.

Here's an example of what I mean

Inside `pages/api` I have a simple `hello.js` 

![image](https://user-images.githubusercontent.com/38945511/160144599-aa074f23-e5e0-44e8-a29e-e560a8ceeea4.png)

When I'm trying to debug the server side next.js endpoints I can only attach breakpoints to a transpiled version that looks like this

![image](https://user-images.githubusercontent.com/38945511/160145024-7fe24b36-525e-4b15-b85f-64a5ecc11d0c.png)

I'm only able to find that file by using `console.log` to give me something in the debug console to link to

**Reply by nandorojo** [reply]
I see, I think you should comment on the expo repo for the next.js integration

**Reply by zksward** [reply]
For now what I might do is have a separate api project that I can use with plain next.js so debugging is simpler. Then I can either use [multi-zones](https://nextjs.org/docs/advanced-features/multi-zones) to stitch the two together or import the handlers from one to the other for usage in production... I'm not sure, build tools and configuration can get really confusing sometimes XD

**Reply by nandorojo** [reply]
Yeah, makes sense. Maybe could have a redirect in your main app's `next.config.js` that redirects `/api/:rest` to `your-other-endpoint.com/api/:rest`. FWIW, with Vercel, any `/api` folder works, even if it's not in a Next.js app.
